Modules/_localemodule.c

```diff
@@ -763,10 +764,58 @@ PyInit__locale(void)
 #endif
 
     if (PyErr_Occurred()) {
-        Py_DECREF(m);
-        return NULL;
+        return -1;
     }
-    return m;
+    return 0;
+}
+
+static struct PyModuleDef_Slot _locale_slots[] = {
+    {Py_mod_exec, _locale_exec},
+    {0, NULL}
+};
+
+static int
+locale_traverse(PyObject *m, visitproc visit, void *arg)
+{
+    _locale_state *state = (_locale_state*)PyModule_GetState(m);
+    if (state) {
```

Comment:
As I asked on your two other PRs (binascii, audioop), I don't think that state can be NULL here. Same remark in locale_clear().

Reply:
Hi, victor. There have some description info about  `traverse slot` in https://docs.python.org/3/c-api/module.html?highlight=pymoduledef#c.PyModuleDef:
```
A traversal function to call during GC traversal of the module object, or NULL if not needed. This function may be called before module state is allocated (PyModule_GetState() may return NULL), and before the Py_mod_exec function is executed.
```

So this behavior is a planned behavior, isn't it?

Reply:
Let's discuss that in https://bugs.python.org/issue39824

Reply:
The discussion won't be over before Nick Coghlan is satisfied (and that's a good thing!)

Meanwhile, this PR is correct according to current behavior and documentation. I don't think bpo-39824 should block it.

Reply:
Alright. I expected https://bugs.python.org/issue39824 to be resolved quick, but I'm ok to merge @shihai1991 PR's first, and revisit the code later if we decided that m_clear/m_free cannot be called with a NULL state.